src/Operations.jl

```diff
@@ -666,5 +667,22 @@ function test(env::EnvCache, pkgs::Vector{PackageSpec}; coverage=false)
                  " errored during testing")
     end
 end
+
+function init(path::String)
+    gitpath = nothing
+    try
+        gitpath = git_discover(path, ceiling = homedir())
+    catch err
+        err isa LibGit2.GitError && err.code == LibGit2.Error.ENOTFOUND || rethrow(err)
+    end
+    path = gitpath == nothing ? path : dirname(dirname(gitpath))
+    try mkpath(path) end
+    for f in ("Project.toml", "Manifest.toml")
+        isfile(joinpath(path, f)) && cmderror("Project already initialized at $path")
+    end
+    touch(joinpath(path, "Project.toml"))
```

Comment:
It's a bit strange to create an empty file here. I wonder if we shouldn't prompt to add some fields with good initial guesses. For example:
```
project = basename(path)
author = "$(readchomp(`git config user.name`)) <$(readchomp(`git config user.email`))>"
```
We could prompt and let the user edit or skip but at least they put something useful in the project file. There's also probably a LibGit2 way to do that instead of shelling out, but you get my point.